interfaces.go

```diff
@@ -146,7 +146,7 @@ type FilterQuery struct {
 	// {{}, {B}}          matches any topic in first position, B in second position
 	// {{A}}, {B}}        matches topic A in first position, B in second position
 	// {{A, B}}, {C, D}}  matches topic (A OR B) in first position, (C OR D) in second position
-	Topics [][]common.Hash
+	Topics [][]*common.Hash
```

Comment:
I don't get why this change is needed. The wildcard for a topic is the nil slice as documented just above this line. `(anything OR XXX)` is the same as `anything`.

Reply:
[][]common.Hash means

[t1 OR t2] AND [t3 or t4]

You can't set t1 or t2 or t3 or t4 to nil, only the entire group.

Reply:
Ah, you mean interpreting the `null`s already during parsing and getting rid of them.